samples/snippets/conftest.py

```diff
@@ -43,6 +45,15 @@ def project_id(bigquery_client: bigquery.Client) -> str:
     return bigquery_client.project
 
 
+@pytest.fixture(autouse=True)
+def reset_session():
+    """An autouse fixture ensuring each sample runs in a fresh session.
+
+    This allows us to have samples that query data in different locations.
+    """
+    bpd.reset_session()
```

Comment:
Apparently, this isn't sufficient to allow location autodetect to work, let's also reset the location:

```
>           raise ValueError(
                f"Current session is in {self._location} but dataset '{table.project}.{table.dataset_id}' is located in {table.location}"
            )
E           ValueError: Current session is in EU but dataset 'bigquery-public-data.ml_datasets' is located in US
```

```suggestion
    bpd.reset_session()
    bpd.options.bigquery.location = None
```